File: gentrificationFactors/DC-Hoods.R
library(stringr)
library(plyr)
library(Hmisc)
### Read in Data ###
### Read in Data ###
### Read in Data ###
hoodRaw<-read.csv('/Users/katerabinowitz/Documents/DataLensDC/DC-Neighborhoods/neighborhooddata.csv',na.strings=c("", "NA"),
                   strip.white=TRUE) 
hoodRaw$income0812<-as.numeric(gsub(",","", hoodRaw$income0812))
hoodRaw$income00<-as.numeric(gsub(",","", hoodRaw$income00))

### Changes over Time ###
### Changes over Time ###
### Changes over Time ###
hoodDiff<-hoodRaw
attach(hoodDiff)
hoodDiff$income<-((income0812-income00)/income00)*100
hoodDiff$bl<-((bl10-bl00)/bl00)*100
hoodDiff$wh<-wh10-wh00
hoodDiff$hs<-hs10-hs00
hoodDiff$emp<-emp0812-emp00
hoodDiff$pCrime<-((pCrime10-pCrime00)/pCrime00)*100
hoodDiff$vCrime<-((vCrime10-vCrime00)/vCrime00)*100
hoodDiff$poverty<-poverty0812-poverty00

hoodDiff$home00<-as.numeric(as.character(gsub(',','',home00)))
hoodDiff$home13<-as.numeric(as.character(gsub(',','',home13)))

hoodDiff$NAME<-paste("Cluster ", hoodDiff$X.built.in.function.id.,sep="")


### Explore Key Aspects of Gentrification ###
### Explore Key Aspects of Gentrification ###
### Explore Key Aspects of Gentrification ###
#income, home price, proportion of minorities and proportion of families/children
hoodDiff<-hoodDiff[order(-hoodDiff$income), ]
describe(hoodDiff)
#subset neighborhoods to those below median income in 2000
hoodBMI<-subset(hoodDiff,hoodDiff$income00<77515)
attach(hoodBMI)
cor(bl, income)
plot(kidCh10,income)

hoodHome<-subset(hoodBMI,!(hoodBMI$home13=='i'))
hoodHome$home<-((hoodHome$home13-hoodHome$home00)/hoodHome$home00)*100
hoodHome<-hoodHome[order(-hoodHome$home), ]

tocsv<-hoodBMI[c('NAME','income','bl','kidCh10')]
tocsv2<-hoodHome[c('NAME','home')]


write.csv(tocsv, 
          file="/Users/katerabinowitz/Documents/DataLensDC/DC-Neighborhoods/gentdemo.csv",row.names=FALSE)
write.csv(tocsv2, 
          file="/Users/katerabinowitz/Documents/DataLensDC/DC-Neighborhoods/genthouse.csv",row.names=FALSE)

File: gentrificationFactors/DC-Hoods.py
import pandas as pd
import numpy as np
from bs4 import BeautifulSoup
import requests 


cluster_id=range(1,40)

popCh10=[]
kidCh10=[]
bl00=[]
bl10=[]
wh00=[]
wh10=[]
hs00=[]
hs10=[]
as00=[]
as10=[]

poverty00=[]
poverty08_12=[]
unemployment00=[]
unemployment08_12=[]
employed00=[]
employed08_12=[]
avgFamIncome00=[]
avgFamIncome08_12=[]
violentCrime00=[]
violentCrime10=[]
propCrime00=[]
propCrime10=[]

homeOwnership00=[]
homeOwnership08_12=[]

medianHomePrice00=[]
medianHomePrice10=[]
medianHomePrice13=[]


for num in cluster_id:
    r = requests.get('http://neighborhoodinfodc.org/nclusters/Nbr_prof_clus' + str(num) + '.html')
    b = BeautifulSoup(r.text)
    popCh10.append(b('table')[2].find_all('tr')[9].find_all('td')[1].text)  
    kidCh10.append(b('table')[2].find_all('tr')[17].find_all('td')[1].text)   
    bl00.append(b('table')[2].find_all('tr')[28].find_all('td')[1].text)
    bl10.append(b('table')[2].find_all('tr')[29].find_all('td')[1].text)
    wh00.append(b('table')[2].find_all('tr')[31].find_all('td')[1].text)
    wh10.append(b('table')[2].find_all('tr')[32].find_all('td')[1].text)
    hs00.append(b('table')[2].find_all('tr')[34].find_all('td')[1].text)
    hs10.append(b('table')[2].find_all('tr')[35].find_all('td')[1].text)
    as00.append(b('table')[2].find_all('tr')[37].find_all('td')[1].text)
    as10.append(b('table')[2].find_all('tr')[38].find_all('td')[1].text)
    
    r = requests.get('http://neighborhoodinfodc.org/nclusters/Nbr_prof_clusb' + str(num) + '.html')
    b = BeautifulSoup(r.text)
    poverty00.append(b('table')[2].find_all('tr')[5].find_all('td')[1].text)
    poverty08_12.append(b('table')[2].find_all('tr')[6].find_all('td')[1].text)
    unemployment00.append(b('table')[2].find_all('tr')[16].find_all('td')[1].text)
    unemployment08_12.append(b('table')[2].find_all('tr')[17].find_all('td')[1].text)
    employed00.append(b('table')[2].find_all('tr')[20].find_all('td')[1].text)
    employed08_12.append(b('table')[2].find_all('tr')[21].find_all('td')[1].text)
    avgFamIncome00.append(b('table')[2].find_all('tr')[35].find_all('td')[1].text)
    avgFamIncome08_12.append(b('table')[2].find_all('tr')[36].find_all('td')[1].text)
    violentCrime00.append(b('table')[2].find_all('tr')[73].find_all('td')[1].text)
    violentCrime10.append(b('table')[2].find_all('tr')[83].find_all('td')[1].text)
    propCrime00.append(b('table')[2].find_all('tr')[89].find_all('td')[1].text)
    propCrime10.append(b('table')[2].find_all('tr')[96].find_all('td')[1].text)
    
    r = requests.get('http://neighborhoodinfodc.org/nclusters/Nbr_prof_clusc' + str(num) + '.html')
    b = BeautifulSoup(r.text)
    medianHomePrice00.append(b('table')[2].find_all('tr')[46].find_all('td')[1].text)
    medianHomePrice10.append(b('table')[2].find_all('tr')[56].find_all('td')[1].text)
    medianHomePrice13.append(b('table')[2].find_all('tr')[59].find_all('td')[1].text)
    
neighborhood_data = pd.DataFrame({id:cluster_id, 'popCh10':popCh10, 'kidCh10':kidCh10, 'bl00':bl00, 'bl10':bl10, 'wh00':wh00, 'wh10':wh10, 'hs00':hs00, 'hs10':hs10,
								'as00':as00, 'as10':as10, 'poverty00':poverty00, 'poverty0812':poverty08_12, 'unemp00':unemployment00,
								'unemp0812':unemployment08_12, 'emp00':employed00, 'emp0812':employed08_12, 'income00':avgFamIncome00,
								'income0812':avgFamIncome08_12, 'vCrime00':violentCrime00, 'vCrime10':violentCrime10, 'pCrime00':propCrime00,
								'pCrime10':propCrime10, 'home00':medianHomePrice00, 'home10':medianHomePrice10, 'home13':medianHomePrice13})

neighborhood_data.to_csv('neighborhooddata.csv')

